packages/near-membrane-dom/src/custom-devtools-formatter.ts

```diff
@@ -0,0 +1,306 @@
+        if (boxedHeaderEntry) {
+            formattedHeader[formattedHeaderOffset++] = boxedHeaderEntry;
+            if (keysLength) {
+                formattedHeader[formattedHeaderOffset++] = ['span', {}, ', '];
+            }
+        }
+        for (let i = 0, length = MathMin(keysLength, MAX_OBJECT_DISPLAY); i < length; i += 1) {
+            const key = keys[i];
+            const value = object[key];
+            formattedHeader[formattedHeaderOffset++] = ['span', {}, i ? ', ' : ''];
+            formattedHeader[formattedHeaderOffset++] = ['span', primitiveGreyColorStyleObject, key];
+            formattedHeader[formattedHeaderOffset++] = ['span', {}, ': '];
+            formattedHeader[formattedHeaderOffset++] = formatValue(value);
+        }
+        if (keysLength > MAX_OBJECT_DISPLAY) {
+            formattedHeader[formattedHeaderOffset++] = [
+                'span',
+                null,
+                ['span', {}, `, ${CHAR_ELLIPSIS}`],
+            ];
+        }
+        formattedHeader[formattedHeaderOffset++] = ['span', {}, '}'];
+        return formattedHeader;
+    };
+    // istanbul ignore next: currently unreachable via tests
+    const formatBody = function formatBody(object: object) {
+        const keys = ObjectKeys(object);
+        // @TODO: Arrays are broken into groups of 100.
+        const ownKeys = ReflectOwnKeys(object);
+        if (!ArrayIsArray(object)) {
+            ReflectApply(ArrayProtoSort, ownKeys, []);
+        }
+        const formattedBody: any[] = [];
+        let formattedBodyOffset = 0;
+        for (let i = 0, { length } = ownKeys; i < length; i += 1) {
+            const ownKey = ownKeys[i];
+            const value = (object as any)[ownKey];
+            if (isObject(value)) {
+                formattedBody[formattedBodyOffset++] = [
+                    'div',
+                    {},
+                    [
+                        'object',
+                        {
+                            object: value,
+                            config: { childKey: StringCtor(ownKey), isChildElement: true },
+                        },
+                    ],
+                ];
+            } else {
+                let currentKeyStyle = keyEnumerableStringStyleObject;
+                if (
+                    typeof ownKey === 'symbol' ||
+                    !ReflectApply(ArrayProtoIncludes, keys, [ownKey])
+                ) {
+                    currentKeyStyle = keyNonEnumerableOrSymbolStyleObject;
+                }
+                formattedBody[formattedBodyOffset++] = [
+                    'div',
+                    bodyItemStyleObject,
+                    ['span', currentKeyStyle, StringCtor(ownKey)],
+                    ['span', {}, ': '],
+                    formatValue(value),
+                ];
+            }
+        }
+        return formattedBody;
+    };
+
+    let { devtoolsFormatters } = window;
```

Comment:
prefer self or globalThis, otherwise it won't work in Web Worker.

Reply:
The current `near-membrane-dom`, where this lives, references `window` and does not support running in a Web Worker.